### ISSUE TITLE
Grails 3.3 url constraint throwing Exception

### ISSUE THREAD

**Issue by philippreston** [grails/grails-core#10804]
### Task List

- [x] Steps to reproduce provided
- [x] Stacktrace (if present) provided
- [x] Example that reproduces the problem uploaded to Github
- [x] Full description of the issue provided (see below)

### Steps to Reproduce

1. Checkout the repository [here](https://github.com/philippreston/url-bug-app) which replicates the issue.  This is a fresh grails 3.3 application.
2. Run `grails run-app` to build and start the app, and browse to http://localhost:8080
3. There is only one Controller (url.bug.app.MissionController) - click it
4. Click "New Mission"
5. Will see the server will throw 500 and display part of the error (full stack trace further below)

<img width="1279" alt="screen shot 2017-09-14 at 22 50 15" src="https://user-images.githubusercontent.com/4885250/30457399-2685b608-999f-11e7-836f-e7e6dbdacae2.png">

### Expected Behaviour

So the area that I believe is causing an issue is the `url: true` constraint which I have added to the model, which looks like this:

```groovy
package url.bug.app

class Mission {

    String website

    static constraints = {
        website url: true
    }
}
```

I expect this to enforce that the String entered in the UI will be validated that it is a correctly formed URL.

### Actual Behaviour

The page throws an error as seen in the screenshot above, and I cannot use url constraint.

The stack trace is as follows:

```
java.lang.IllegalArgumentException: Parameter for constraint [url] of property [website] of class [class url.bug.app.Mission] must be a boolean, string, or list value
        at org.grails.validation.UrlConstraint.setParameter(UrlConstraint.java:68)
        at grails.validation.ConstrainedProperty.applyConstraint(ConstrainedProperty.java:1029)
        at org.grails.validation.ConstrainedDelegate.asType(ConstrainedDelegate.groovy:185)
        at grails.plugin.formfields.BeanPropertyAccessorFactory.resolveConstraints(BeanPropertyAccessorFactory.groovy:82)
        at grails.plugin.formfields.BeanPropertyAccessorFactory.resolvePropertyFromPathComponents(BeanPropertyAccessorFactory.groovy:74)
        at grails.plugin.formfields.BeanPropertyAccessorFactory.resolvePropertyFromPath(BeanPropertyAccessorFactory.groovy:59)
        at grails.plugin.formfields.BeanPropertyAccessorFactory.accessorFor(BeanPropertyAccessorFactory.groovy:46)
        at grails.plugin.formfields.FormFieldsTagLib.resolveProperty(FormFieldsTagLib.groovy:374)
        at grails.plugin.formfields.FormFieldsTagLib$_closure3.doCall(FormFieldsTagLib.groovy:157)
        at org.grails.taglib.TagOutput.captureTagOutput(TagOutput.java:78)
        at grails.plugin.formfields.FormFieldsTagLib$_closure2.doCall(FormFieldsTagLib.groovy:116)
        at org.grails.gsp.GroovyPage.invokeTagLibClosure(GroovyPage.java:439)
        at org.grails.gsp.GroovyPage.invokeTag(GroovyPage.java:364)
        at Users_ppreston_Developer_Projects_Groovy_url_bug_app_grails_app_views_mission_create_gsp$_run_closure2$_closure24.doCall(Users_ppreston_Developer_Projects_Groovy_url_bug_app_grails_app_views_mission_create_gsp:74)
        at org.grails.taglib.TagBodyClosure.executeClosure(TagBodyClosure.java:200)
        at org.grails.taglib.TagBodyClosure.captureClosureOutput(TagBodyClosure.java:102)
        at org.grails.taglib.TagBodyClosure.call(TagBodyClosure.java:213)
        at org.grails.plugins.web.taglib.FormTagLib$_closure9.doCall(FormTagLib.groovy:451)
        at org.grails.gsp.GroovyPage.invokeTagLibClosure(GroovyPage.java:446)
        at org.grails.gsp.GroovyPage.invokeTag(GroovyPage.java:364)
        at Users_ppreston_Developer_Projects_Groovy_url_bug_app_grails_app_views_mission_create_gsp$_run_closure2.doCall(Users_ppreston_Developer_Projects_Groovy_url_bug_app_grails_app_views_mission_create_gsp:79)
        at org.grails.taglib.TagBodyClosure.executeClosure(TagBodyClosure.java:200)
        at org.grails.taglib.TagBodyClosure.captureClosureOutput(TagBodyClosure.java:102)
        at org.grails.taglib.TagBodyClosure.call(TagBodyClosure.java:213)
        at org.grails.plugins.web.taglib.SitemeshTagLib.captureTagContent(SitemeshTagLib.groovy:48)
        at org.grails.plugins.web.taglib.SitemeshTagLib$_closure3.doCall(SitemeshTagLib.groovy:156)
        at org.grails.gsp.GroovyPage.invokeTagLibClosure(GroovyPage.java:446)
        at org.grails.gsp.GroovyPage.invokeTag(GroovyPage.java:364)
        at Users_ppreston_Developer_Projects_Groovy_url_bug_app_grails_app_views_mission_create_gsp.run(Users_ppreston_Developer_Projects_Groovy_url_bug_app_grails_app_views_mission_create_gsp:82)
        at org.grails.gsp.GroovyPageWritable.doWriteTo(GroovyPageWritable.java:162)
        at org.grails.gsp.GroovyPageWritable.writeTo(GroovyPageWritable.java:82)
        at org.grails.web.servlet.view.GroovyPageView.renderTemplate(GroovyPageView.java:76)
        at org.grails.web.servlet.view.AbstractGrailsView.renderWithinGrailsWebRequest(AbstractGrailsView.java:71)
        at org.grails.web.servlet.view.AbstractGrailsView.renderMergedOutputModel(AbstractGrailsView.java:55)
        at org.springframework.web.servlet.view.AbstractView.render(AbstractView.java:303)
        at org.grails.web.sitemesh.GrailsLayoutView.renderInnerView(GrailsLayoutView.java:150)
        at org.grails.web.sitemesh.GrailsLayoutView.obtainContent(GrailsLayoutView.java:128)
        at org.grails.web.sitemesh.GrailsLayoutView.renderTemplate(GrailsLayoutView.java:63)
        at org.grails.web.servlet.view.AbstractGrailsView.renderWithinGrailsWebRequest(AbstractGrailsView.java:71)
        at org.grails.web.servlet.view.AbstractGrailsView.renderMergedOutputModel(AbstractGrailsView.java:55)
        at org.springframework.web.servlet.view.AbstractView.render(AbstractView.java:303)
        at org.springframework.web.servlet.DispatcherServlet.render(DispatcherServlet.java:1286)
        at org.springframework.web.servlet.DispatcherServlet.processDispatchResult(DispatcherServlet.java:1041)
        at org.springframework.web.servlet.DispatcherServlet.doDispatch(DispatcherServlet.java:984)
        at org.springframework.web.servlet.DispatcherServlet.doService(DispatcherServlet.java:901)
        at org.springframework.web.servlet.FrameworkServlet.processRequest(FrameworkServlet.java:970)
        at org.springframework.web.servlet.FrameworkServlet.doGet(FrameworkServlet.java:861)
        at org.springframework.web.servlet.FrameworkServlet.service(FrameworkServlet.java:846)
        at org.springframework.boot.web.filter.ApplicationContextHeaderFilter.doFilterInternal(ApplicationContextHeaderFilter.java:55)
        at org.grails.web.servlet.mvc.GrailsWebRequestFilter.doFilterInternal(GrailsWebRequestFilter.java:77)
        at org.grails.web.filters.HiddenHttpMethodFilter.doFilterInternal(HiddenHttpMethodFilter.java:67)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
        at java.lang.Thread.run(Thread.java:748)
```

This relates to the following code in `org.grails.datastore.gorm.validation.constraints.UrlConstraint.java` where the `constraintParameter` is of type `UrlValidator`

```java
    @Override
    protected Object validateParameter(Object constraintParameter) {
        RegexValidator domainValidator;

        if (constraintParameter instanceof Boolean) {
            domainValidator = null;
        }
        else if (constraintParameter instanceof String) {
            domainValidator = new RegexValidator((String) constraintParameter);
        }
        else if (constraintParameter instanceof List<?>) {
            List<?> regexpList = (List<?>) constraintParameter;
            domainValidator = new RegexValidator(regexpList.toArray(new String[regexpList.size()]));
        }
        else {
            throw new IllegalArgumentException("Parameter for constraint [" + ConstrainedProperty.URL_CONSTRAINT +
                    "] of property [" + constraintPropertyName + "] of class [" +
                    constraintOwningClass + "] must be a boolean, string, or list value");
        }

        UrlValidator validator = new UrlValidator(domainValidator,
                UrlValidator.ALLOW_ALL_SCHEMES + UrlValidator.ALLOW_2_SLASHES);


        return validator;
    }
```

### Environment Information

- **Operating System**: MacOS 10.12.5
- **Grails Version:** 3.3.0
- **JDK Version:** 1.8.0_131
- **Groovy Version:** 2.4.11
- **Container Version (If Applicable):** N/a

### Example Application

Please see: https://github.com/philippreston/url-bug-app

**Comment by Leward** [comment]
Hi I think I encountered the same problem, see: https://github.com/grails-fields-plugin/grails-fields/issues/241

Normally, version 3.3.1 should resolve the problem.

**Comment by philippreston** [comment]
@Leward @graemerocher 

This is not resolved by 3.3.1.BUILD-SNAPSHOT

You can use my recreation (see link above) and see that the same error that I reported.  

Can you please reopen the issue.

Many thanks

**Comment by philippreston** [comment]
Looks good in my recreation app. I’ll roll in and test in my app - but I’m sure this should be good. 

Many thanks. 

Phil
